src/internals/color.js

```diff
@@ -62,7 +64,7 @@ extend(ChartInternal.prototype, {
 
 		return function(d) {
 			const id = d.id || (d.data && d.data.id) || d;
-			const isLine = $$.isTypeOf(id, "line");
+			const usePattern = $$.config.data_types[id] !== undefined;
```

Comment:
This can't cover all the "line" types.
"Line" relative types are: line, spline, step and undefined(which is default to line).

So, I think the condition should be as follows.
```js
const isLine = $$.isTypeOf(id, ["line", "spline", "step"]) || !$$.config.data_types[id];
...
color = isLine ?
		originalColorPattern[ids.indexOf(id) % originalColorPattern.length] :
		pattern[ids.indexOf(id) % pattern.length];
```

Reply:
Yes that seems to cover all line types, thanks!